### PR TITLE
shop-manager-challenge complete 

### PR DESCRIPTION
**Pull request by ritaaktay**
DATABASE_NAME and DATABASE_HOST are configured in .env for PG connection

### CODE REVIEW

File: lib/application.rb [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
-      when '5'
-        return
-      end
-    end
+    menu
+    input = @io.gets.chomp
+    print_all_items if input == '1'
+    add_item if input == '2'
+    print_all_orders if input == '3'
+    add_order if input == '4'
```
Comment (ddrmv) [inline lib/application.rb:15]: Nice and short, I personally prefer using case, as it's even more readable at a glance.

File: lib/item.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Item
+  def initializer(name:, price:, stock:)
+    @name = name
+    @price = price
+    @stock = stock
+  end
```
Comment (ddrmv) [inline lib/item.rb:6]: This part seems unnecessary, as attr_accessor does this for us, I believe.

File: spec/order_repository_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+require 'item_repository'
+require 'order_repository'
+require 'item'
+require 'order'
+
+RSpec.describe 'OrderRepository' do
+    describe '#create' do
+      it 'creates a new order' do
```
Comment (ddrmv) [inline spec/order_repository_spec.rb]: I like that it follows convention well.

File: lib/application.rb [hunk 4]
```diff
@@ -60,21 +51,31 @@ def print_item(item)
...
+      stock: stock
+    )
     @item_repository.create(item)
+    @io.puts 'Item created.'
   end
 
   def prompt(message)
     @io.puts message
     @io.gets.chomp
   end
```
Comment (ddrmv) [inline lib/application.rb:80]: Very nice, I like this a lot.

File: spec/application_spec.rb [hunk 5]
```diff
@@ -2,4 +2,106 @@
...
+  end
+
+  before(:each) do
+    reset_tables
+  end
+  
+  describe '#run' do
+    context 'when menu item 1 is seleceted' do
+      it 'lists all items' do
+        menu
```
Comment (ddrmv) [inline spec/application_spec.rb:29]: Nice.

File: spec/application_spec.rb [hunk 6]
```diff
@@ -2,4 +2,106 @@
...
+        expect(io).to receive(:gets).and_return '4'
+        expect(io).to receive(:puts).with('Customer name:').ordered
+        expect(io).to receive(:gets).and_return('Bugs Bunny').ordered
+        expect(io).to receive(:puts).with('Order date:').ordered
+        expect(io).to receive(:gets).and_return('20/08/2022').ordered
+        expect(io).to receive(:puts).with('Item ID:').ordered
+        expect(io).to receive(:gets).and_return('1').ordered
+        expect(io).to receive(:puts).with('Order created.').ordered
+        app.run
+        orders = OrderRepository.new.all_with_item
```
Comment (ddrmv) [inline spec/application_spec.rb:93]: I like the syntax, saves a line, if maybe a tiny bit less readable.

File: lib/application.rb (hunk 1 above)
Comment (ritaaktay) [reply lib/application.rb:15]: Yes I did use case at first as well, then the rubocop told me my method had too many lines, and I figured this to reduce number of lines. I agree that case is much more readable!

File: lib/item.rb (hunk 2 above)
Comment (ritaaktay) [reply lib/item.rb:6]: I wanted to be able to shorten the code when I am creating new Orders or Items.

**Pull request closed without merging**